api/v1beta1/cluster_types.go

```diff
@@ -509,8 +509,8 @@ func (c *Cluster) SetConditions(conditions Conditions) {
 }
 
 // GetIPFamily returns a ClusterIPFamily from the configuration provided.
-// Note: IPFamily is not a concept in Kubernetes. It was originally introduced in CAPI for CAPD.
-// IPFamily may be dropped in a future release. More details at https://github.com/kubernetes-sigs/cluster-api/issues/7521
+// Deprecated: IPFamily is not a concept in Kubernetes. It was originally introduced in CAPI for CAPD.
```

Comment:
```suggestion
//
// Deprecated: IPFamily is not a concept in Kubernetes. It was originally introduced in CAPI for CAPD.
```

Afaik, otherwise (at least) staticcheck won't recognize the deprecation